variables.tf

```diff
@@ -137,3 +139,38 @@ variable "prefixes_pull_through_repositories" {
   description = "Organization IDs to provide with push access to the ECR"
   default     = []
 }
+
+variable "custom_lifecycle_rules" {
+  description = "Custom lifecycle rules to override or complement the default ones"
+  type = list(object({
+    rulePriority = number
```

Comment:
We an reset priority as it is calculated here https://github.com/cloudposse/terraform-aws-ecr/blob/8a08b2632f35c440ee7bd4ae15c968c1f3f205c8/main.tf#L138 
```suggestion
```